### PR TITLE
Add a favicon to the docs and the conda-store server UI

### PR DESCRIPTION
**Pull request by asmeurer**
This makes it easier to find conda-store in your Chrome tabs.

### CODE REVIEW

File: conda-store-server/conda_store_server/server/static/conda-store-ui/favicon.ico
Comment (asmeurer) [inline conda-store-server/conda_store_server/server/static/conda-store-ui/favicon.ico:1]: Is this the correct place to put this file?

Comment (costrouc) [reply conda-store-server/conda_store_server/server/static/conda-store-ui/favicon.ico:1]: I'd prefer if it were added to just the static path. `conda-store-server/conda_store_server/server/static/favicon.ico`. The conda-store-ui directory things get thrown in there from the hatch build.

**Review by costrouc — CHANGES_REQUESTED** [review]
Other than that small fix this looks great to me! Thanks @asmeurer. 

Could you rebase this PR on main now? All tests should be passing.

**Comment by costrouc** [comment]
@asmeurer I force pushed to rebase this PR on `main`.

**Review by costrouc — APPROVED** [review]

**Pull request merged**